pkg/services/alerting/alert_validator.go

```diff
@@ -0,0 +1,15 @@
+package alerting
+
+import "github.com/grafana/grafana/pkg/models"
+
+type validatorSeverity int
+
+const (
+	alertWarning validatorSeverity = iota
+	alertError
+)
+
+type alertValidator struct {
+	aFunc     func(*models.Alert) (bool, string)
```

Comment:
Don't see any point in returning also a bool, since that should correspond to whether the reason is empty or not:
```suggestion
	aFunc     func(*models.Alert) string
```